gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1179,46 +1200,81 @@ reconcile_when_close_toggled_cb (GtkToggleButton *togglebutton, ofx_info* info)
     info->run_reconcile = gtk_toggle_button_get_active (togglebutton);
 }
 
+static gchar* make_date_amount_key (time64 date, gnc_numeric amount)
+{
+    // Create a string that combines date and amount, we'll use that for our hash
+    gchar buf[64];
+    gnc_numeric _amount = gnc_numeric_reduce(amount);
+    g_snprintf (buf, sizeof(buf), "%" PRId64 "%" PRId64 "%" PRId64, _amount.num , _amount.denom, date);
+    return g_strdup (buf);
+}
+
 static void
-runMatcher(ofx_info* info, char * selected_filename, gboolean go_to_next_file)
+runMatcher (ofx_info* info, char * selected_filename, gboolean go_to_next_file)
 {
     GtkWindow *parent = info->parent;
     GList* trans_list_remain = NULL;
     Account* first_account = NULL;
 
     /* If we have multiple accounts in the ofx file, we need to
-     * process transactions one account at a time, in case there are
-     * transfers between accounts.
+     * avoid processing transfers between accounts together because this will
+     * create duplicate entries.
      */
+    GHashTable* trans_hash = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                    g_free, NULL);
     info->num_trans_processed = 0;
+    // Add transactions, but verify that there isn't one that was already added with identical
+    // amounts and date, and a different account. To do that, create a hash table whose key is
+    // a hash of amount and date, and whose value is the account in which they appear.
     for(GList* node = info->trans_list; node; node=node->next)
     {
         Transaction* trans = node->data;
         Split* split = xaccTransGetSplit (trans, 0);
-        if (first_account == NULL) first_account = xaccSplitGetAccount (split);
-        if (xaccSplitGetAccount (split) == first_account)
+        Account* account = xaccSplitGetAccount (split);
+        gchar *date_amount_key = make_date_amount_key (xaccTransGetDate (trans),
+                                                      gnc_numeric_abs (xaccSplitGetAmount (split)));
+        // Test if date_amount_key is already in trans_hash.
+        Account* _account = g_hash_table_lookup (trans_hash, date_amount_key);
+        if (_account && _account != account)
+        {
+            // There is a transaction with identical amounts and
+            // dates, but a different account.  That's a potential
+            // transfer so process this transaction in a later call.
+            gchar *name1 = gnc_account_get_full_name (account);
+            gchar *name2 = gnc_account_get_full_name (_account);
+            gchar *amtstr = gnc_numeric_to_string (xaccSplitGetAmount (split));
+            gchar *datestr = qof_print_date (xaccTransGetDate (trans));
+            DEBUG ("Potential transfer %s %s %s %s\n", name1, name2, amtstr, datestr);
+            g_free (name1);
+            g_free (name2);
+            g_free (amtstr);
+            g_free (datestr);
```

Comment:
@jralls is there a run-time directive we can use to test whether DEBUG is active? All these allocations could be skipped when DEBUG is disabled...

Reply:
There is:
```
if (qof_log_check (logdomain, QOF_LOG_DEBUG)
{
    /* Make the debug args */
    DEBUG (/* diagnostic */);
}
```